platform/ui/src/components/StepProgressDropdown/StepProgressDropdown.tsx

```diff
@@ -0,0 +1,326 @@
+import React, {
+  ReactNode,
+  useEffect,
+  useCallback,
+  useState,
+  useMemo,
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Icon, Tooltip } from '../';
+
+const MAX_TOOLTIP_LENGTH = 250;
+const DROPDOWN_OPTION_PROPTYPE = PropTypes.shape({
+  label: PropTypes.string.isRequired,
+  value: PropTypes.string.isRequired,
+  info: PropTypes.string,
+  activated: PropTypes.bool,
+  completed: PropTypes.bool,
+  onSelect: PropTypes.func,
+});
+
+type StepProgressDropdownOption = {
+  label: string;
+  value: string;
+  info?: string;
+  activated?: boolean;
+  completed?: boolean;
+  onSelect?: () => void;
+};
+
+const StepProgressStatus = ({ options }) => {
```

Comment:
please extract/refactor this code to have each function/component in a separate file for readability